File: app/src/main/java/rod/bailey/trafficatnsw/ui/PlaceholderMessageView.java
package rod.bailey.trafficatnsw.ui;

import android.content.Context;
import android.widget.TextView;

import static android.graphics.Color.*;
import static android.graphics.Typeface.*;
import static android.text.TextUtils.TruncateAt.END;
import static android.view.Gravity.*;
import static rod.bailey.trafficatnsw.util.DisplayUtils.*;

public class PlaceholderMessageView extends TextView {
	
	private static final int TEXT_SIZE_SP = 14;

	public PlaceholderMessageView(Context ctx) {
		super(ctx);
		
		setTextColor(GRAY);
		setTypeface(DEFAULT);
		setBackgroundColor(viewBackgroundColor());
		setGravity(CENTER);
		setTextSize(TEXT_SIZE_SP);
		setClickable(false);
		setEllipsize(END);
	}

	public PlaceholderMessageView(Context ctx, String message) {
		this(ctx);
		setMessage(message);
	}
	
	public void setMessage(String msg) {
		setText(msg);
	}

}

File: app/src/main/java/rod/bailey/trafficatnsw/ui/view/PlaceholderMessageView.kt
package rod.bailey.trafficatnsw.ui.view

import android.content.Context
import android.support.v7.widget.AppCompatTextView
import android.view.LayoutInflater
import android.view.View
import android.widget.FrameLayout
import rod.bailey.trafficatnsw.R

class PlaceholderMessageView(ctx: Context, message: String) : FrameLayout(ctx) {

	init {
		val inflater: LayoutInflater = LayoutInflater.from(ctx)
		val content: View = inflater.inflate(R.layout.view_empty_list_placeholder, this)
		val textView: AppCompatTextView = content.findViewById(R.id.tv_empty_list_placeholder) as AppCompatTextView
		textView.text = message
	}

}
